### PR TITLE
Final on Uri Parsing

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
Allowed both modes (backtick enclosing and/or url_encoding).
Modified and integrated documentation.

Added Minor log fix.
When solution is found also output the job it refers.

### CODE REVIEW

**Comment by StefanOberhumer** [comment]
Instead
```
// Now let's see if authority part can be split into userinfo and "the rest"
std::regex usr_url("^(.*)\\@(.*)$");
```

shouldn't we us the **non greedy** version ?

```
// Now let's see if authority part can be split into userinfo and "the rest"
std::regex usr_url("^(.*?)\\@(.*?)$");
```

That would allow us to use
`-P stratum1+tcp://0x9E431042fAA3224837e9BEDEcc5F4858cf0390B9@eth-eu1.nanopool.org:9999/pooltester/andrea.lanfranchi@gmail.com`
instead
`-P stratum1+tcp://0x9E431042fAA3224837e9BEDEcc5F4858cf0390B9@eth-eu1.nanopool.org:9999/pooltester/andrea.lanfranchi%40gmail.com`

**Comment by AndreaLanfranchi** [comment]
It would fail in the other case ... when there is an unencoded `@` among ticks.
https://regex101.com/r/JywIDY/1/

**Comment by AndreaLanfranchi** [comment]
It might work if I tokenize all authority part (m_authority) instead of only the m_userinfo

File: libethash-cl/CLMiner.cpp [hunk 1]
```diff
@@ -398,7 +398,10 @@ void CLMiner::workLoop()
                         memcpy(mix.data(), (char*)results.rslt[i].mix, sizeof(results.rslt[i].mix));
                         auto sol = Solution{
                             nonce, mix, current, std::chrono::steady_clock::now(), m_index};
-                            cllog << "Sol: " << EthWhite "0x" << toHex(sol.nonce) << EthReset;
+                        
+                        cllog << EthWhite << "Job: " << w.header.abridged() << " Sol: 0x"
```
Comment (jean-m-cyr) [inline libethash-cl/CLMiner.cpp]: Maybe add a # in front of the job id, to be consistent with other log entries...

Or, remove the # from other log jobids.

Comment (AndreaLanfranchi) [reply libethash-cl/CLMiner.cpp]: Ok done.

**Comment by AndreaLanfranchi** [comment]
@StefanOberhumer I understand objections.
Will withdraw this PR. Maybe other lighter solutions will be found.

**Comment by StefanOberhumer** [comment]
@AndreaLanfranchi : I take that part - THX for your spent time. 
I started testing with scripts/testpools.bash - seems a some more pools does not work a longer time....

**Pull request closed without merging**